server/users.ts

```diff
@@ -902,12 +903,14 @@ export class User extends Chat.MessageContext {
 				this.autoconfirmed = userid;
 			} else if (userType === '4') {
 				this.autoconfirmed = userid;
-			} else if (userType === '5') {
+			} else if (userType === '5' || userType === '6') {
 				this.permalocked = userid;
-				Punishments.lock(this, Date.now() + PERMALOCK_CACHE_TIME, userid, true, `Permalocked as ${name}`);
-			} else if (userType === '6') {
-				Punishments.lock(this, Date.now() + PERMALOCK_CACHE_TIME, userid, true, `Permabanned as ${name}`);
-				this.disconnectAll();
+				if (this.latestHostType === 'mobile') {
+					Punishments.lock(this, Date.now() + PERMALOCK_MOBILE_CACHE_TIME, userid, true, `Permalocked as ${name}`);
```

Comment:
This should very likely be handled inside `Punishments.lock` rather than outside.

Reply:
Especiallly since the userid and other IP locks should be for the full time; only the mobile IP lock should be short.

And all mobile IP locks should be short, not just permalocks.